### DISCUSSION TITLE
Where is `undefined` module🤔

### DISCUSSION THREAD

**Discussion by kloczek** [jmcgeheeiv/pyfakefs#981] [Q&A]
Looks like test suite uses `undefined` modules.
All what I found on pypy is old and umacniane.
<details>
<summary>Here is pytest output:</summary>

```console
+ /usr/bin/pytest -ra -m 'not network' --import-mode=importlib
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.9.18, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pyfakefs-5.3.5
configfile: pyproject.toml
plugins: pyfakefs-5.3.5, hypothesis-6.99.5, xdist-3.5.0, forked-1.6.0
collected 2168 items / 1 error

========================================================================================== ERRORS ===========================================================================================
_______________________________________________________________ ERROR collecting pyfakefs/pytest_tests/pytest_fixture_test.py _______________________________________________________________
ImportError while importing test module '/home/tkloczko/rpmbuild/BUILD/pyfakefs-5.3.5/pyfakefs/pytest_tests/pytest_fixture_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
pyfakefs/pytest_tests/pytest_fixture_test.py:16: in <module>
    import undefined
E   ModuleNotFoundError: No module named 'undefined'
================================================================================== short test summary info ==================================================================================
ERROR pyfakefs/pytest_tests/pytest_fixture_test.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
===================================================================================== 1 error in 1.39s ======================================================================================
```
</details>

**Comment by mrbean-bremen** [comment]
It's indeed used for a regression test. ~Should be skipped if not installed, I will add this...~
It can actually just be replaced, it just defines a very simple object. Will do this.

**Comment by kloczek** [comment]
> It can actually just be replaced, it just defines a very simple object. Will do this.

Simple both versions available on pypi are unmaintained and hard to package on modern python 3.x (with testing/generate documentation etc).
If it is anything which is replacement and is well maintained please do that 😋 
May I ask what you are going to use? (will try to package that in advance 😄 )

**Reply by mrbean-bremen** [reply]
As I wrote, I will just add a simple module to the test, so that there will be no need for a package.
Same as with the other issue - will do this after I finished with a few other open issues, so may take some time.

**Reply by kloczek** [reply]
OK .. but you wrote:
> It can actually just be replaced, it just defines a very simple object. Will do this.

Does it mean that that necessary code will be added directly in this module?  😋

**Reply by mrbean-bremen** [reply]
Yes. As this is a trivial change, I'm doing it right now (_Edit_: done),  the release will wait until the other issues are fixed, though.

**Reply by kloczek** [reply]
OK .. will wait 😋 
Please let me know if you will have some change.
I'll be happy to test that 👍 😄

**Reply by mrbean-bremen** [reply]
A [new release](https://pypi.org/project/pyfakefs/5.4.0) is out now.
